miner/worker.go

```diff
@@ -737,20 +740,33 @@ func (w *worker) resultLoop() {
 
 // makeEnv creates a new environment for the sealing block.
 func (w *worker) makeEnv(parent *types.Header, header *types.Header, coinbase common.Address) (*environment, error) {
-	// Retrieve the parent state to execute on top and start a prefetcher for
-	// the miner to speed block sealing up a bit.
-	state, err := w.chain.StateAt(parent.Root)
-	/* [kroma unsupported]
-	if err != nil && w.chainConfig.Optimism != nil { // Allow the miner to reorg its own chain arbitrarily deep
-		if historicalBackend, ok := w.eth.(BackendWithHistoricalState); ok {
-			var release tracers.StateReleaseFunc
-			parentBlock := w.eth.BlockChain().GetBlockByHash(parent.Hash())
-			state, release, err = historicalBackend.StateAtBlock(context.Background(), parentBlock, ^uint64(0), nil, false, false)
-			state = state.Copy()
-			release()
+	var state *state.StateDB
+	var err error
+
+	// [Kroma: ZKT to MPT]
+	if w.chainConfig.IsKromaMPTActivationBlock(header.Time) {
+		w.chainConfig.Zktrie = false
+		w.chain.TrieDB().SetBackend(false)
```

Comment:
It seems to be beyond the scope of the makeEnv function. What if we separate the logic separately?